frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/PolicyAccessPackages.tsx

```diff
@@ -0,0 +1,164 @@
+import { usePolicyRuleContext } from '../../../../contexts/PolicyRuleContext';
+import classes from './PolicyAccessPackages.module.css';
+import { PolicyAccessPackageAccordion } from './PolicyAccessPackageAccordion';
+import { PolicyAccordion } from './PolicyAccordion/PolicyAccordion';
+import { groupAccessPackagesByArea } from '@altinn/policy-editor/utils';
+
+const CHECKED_VALUE = 'on';
+const selectedLanguage = 'nb';
+
+export const PolicyAccessPackages = (): React.ReactElement => {
+  const { t } = useTranslation();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const { policyRules, accessPackages, setPolicyRules, savePolicy } = usePolicyEditorContext();
+  const { policyRule } = usePolicyRuleContext();
+
+  const [chosenAccessPackages, setChosenAccessPackages] = useState<string[]>(
+    policyRule.accessPackages,
+  );
+
+  const groupedAccessPackagesByArea = useMemo(() => {
+    return groupAccessPackagesByArea(accessPackages);
+  }, [accessPackages]);
+
+  const onPackageSelectChange = (accessPackage: PolicyAccessPackage): void => {
+    const isSelected = chosenAccessPackages.includes(accessPackage.urn);
+    if (isSelected) {
+      handleRemoveAccessPackage(accessPackage.urn);
+    } else {
+      handleAddAccessPackage(accessPackage.urn);
+    }
+  };
+
+  const handleRemoveAccessPackage = (packageUrn: string): void => {
+    setChosenAccessPackages((oldUrns) => oldUrns.filter((urn) => urn !== packageUrn));
+    const urnsToSave = policyRule.accessPackages.filter((x) => x !== packageUrn);
+
+    handleAccessPackageChange(urnsToSave);
+  };
+
+  const handleAddAccessPackage = (packageUrn: string): void => {
+    setChosenAccessPackages((oldUrns) => [...oldUrns, packageUrn]);
+    const urnsToSave = [...policyRule.accessPackages, packageUrn];
+
+    handleAccessPackageChange(urnsToSave);
+  };
+
+  const handleAccessPackageChange = (newSelectedAccessPackageUrns: string[]): void => {
+    const updatedRules = getUpdatedRules(
+      {
+        ...policyRule,
+        accessPackages: newSelectedAccessPackageUrns,
+      },
+      policyRule.ruleId,
+      policyRules,
+    );
+
+    setPolicyRules(updatedRules);
+    savePolicy(updatedRules);
+  };
+
+  const handleSearch = (search: string) => {
+    setSearchValue(search);
+  };
+
+  const isStringMatch = (matchString: string) => {
+    return matchString.toLowerCase().includes(searchValue.toLowerCase());
+  };
+
+  const accessPackagesToRender = groupedAccessPackagesByArea.reduce(
+    (areas: PolicyAccessPackageArea[], area): PolicyAccessPackageArea[] => {
+      const matchingPackages = area.packages.filter(
+        (pack) => !searchValue || isStringMatch(pack.name) || isStringMatch(pack.description),
+      );
+      const returnAreas = [...areas];
+      if (matchingPackages.length > 0) {
+        returnAreas.push({ ...area, packages: matchingPackages });
+      }
+      return returnAreas;
+    },
+    [],
+  );
```

Comment:
This is a lot of helper functions. Can they be moved to a utils file at same level as this component? 🤔